var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

```diff
@@ -71,11 +72,12 @@ class PyPennylaneLightningKokkos(CMakePackage, PythonExtension, CudaPackage, ROC
     depends_on("py-pybind11", type="link")
     depends_on("py-pip", type="build")
     depends_on("py-wheel", type="build")
-    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@0.31:")
     depends_on("py-pennylane@0.28:", type=("build", "run"), when="@:0.30.0")
-    depends_on("py-pennylane-lightning@0.31:~kokkos", type=("build", "run"), when="@0.31.0:")
-    depends_on("py-pennylane-lightning@0.30:~kokkos", type=("build", "run"), when="@0.30.0:")
-    depends_on("py-pennylane-lightning@0.28:0.29~kokkos", type=("build", "run"), when="@0.29.1")
+    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@0.31:")
```

Comment:
In line 76, I see a version mismatch, `py-pennylane@0.30:` vs `when="@0.31:".

Is this intended?

Reply:
Yes, PennyLane and Lightning have a circular dependency. This may not be a problem in `spack`, but we usually pick a side and have Lightning depend on the previous PennyLane release.